src/services/exposed/record-serializer.js

```diff
@@ -8,9 +8,7 @@ class RecordSerializer extends AbstractRecordService {
     if (!(model instanceof Object)) {
       throw new Error('RecordSerializer initialization error: "model" argument should be an object (ex: `{ name: "myModel" }`)');
     }
-    if (!model.modelName) {
-      model.modelName = model.name;
-    }
```

Comment:
You should consider re-adding theses lines since it'll break smart collection. Not sure why this fix was introduced here though, but on my side it does not work :/